# Circtools Modules Documentation


#### General Workflow of Circtools 
    1. Detect 
    2. Quickcheck 
    3. CircTest
    4. Reconstruct
    5. Exon
    6. Enrich 
    7. Primer 
    ![Alt text](<Screenshot 2023-11-21 at 10.55.33 AM.png>)

## Detect Module 

The `detect` command is an interface to DCC, a python package intended to detect and quantify circRNAs with high specify from RNA sequencing data. This module is foundational to all other steps for the circtools work flow. 

#### General Usage

The detection of circRNA from RNA sequencing data utilizing the `detect` module can be summarized in a few steps: 

- Mapping of RNAseq data from quality checked FASTQ files. For paired-end (PE) data it is recommended to map the pair jointly and separately as STAR does not output reads or read pairs that contain more tha one chimeric junction
- Prepare the input files required by DCC with the mandatory `samplesheet` file which specifies the locations for the `chimeric.out.junction` files (one relative or absolute path per line). [Command line flag: `@samplesheet`]
- A GTF format annotation of repetitive regions which is used to filter out circRNA candidates from repetitive regions. [Command line flag: `-R your_repeat_file.gtf`]
- For paired-end sequencing two files, e.g. `mate1` and `mate2` which contain the paths to the `chimeric.out.junction` files originating from the separate mate mapping step. [Command line flags: `-m1 mate1file` and `-m2 mate2file`]
- You may specify the location of the BAM files via the -B flag otherwise DCC tries to guess their location based on the supplied `chimeric.out.junction` paths. [Command line flag: `-B @bam_file_list`]
- DCC requires the `SJ.out.tab` files generated by STAR. DCC assumes they are located in the same folder as the BAM files and also retain their `SJ.out.tab`.
- DCC can be used to process circular RNA detection and host gene expression detection either in a one-pass strategy or might be used for only one part of the analysis:
            - Detect circRNAs and host gene expression: specify `-D` and `-G` option 
            - Detect circRNAs only: `-D`
            - Detect host gene expression only: `-G` option 


#### Step-by-Step 
A tutorial 


## Quick Check Module 

The `circtools quickcheck` module is a simple way to check the results of a DCC run to assess the quality and number of circRNAs in a given experiment. This module needs the mapping log files produced by STAR as well as the directory with the DCC results. The module will produce a series of figures in PDF format to assess the results 

#### Required tools and packages 

`quickcheck` depends on on R and two R packages, namely

    - ggplot2: general plotting

    - ggrepel: label assignment in plots

#### General usage 

A call to `circtools quickcheck` --help` shows all available command line flags 

```bash
usage: circtools [-h] -d DETECT_DIR -s STAR_DIR -l CONDITION_LIST -g GROUPING
                 [-o OUTPUT_DIRECTORY] [-n OUTPUT_NAME] [-c {colour,bw}]
                 [-C CLEANUP] [-S STARFOLDER] [-L REMOVE_SUFFIX_CHARS]
                 [-F REMOVE_PREFIX_CHARS] [-R REMOVE_COLUMNS]
```

##### Sample Call
*Call*
```bash
circtools quickcheck -d 01_detect/ -s ../star  -l minus,plus -g 1,2,1,2,1,2,1,2  -o 02_quickcheck/  -C .Chimeric.out.junction
```

## Reconstruct module 

The `reconstruct` command is an interface to FUCHS (FUll circular RNA CHaracterization from RNA-Seq), a Python program designed to fully characterize circular RNAs. It uses a list of circular RNAs and reads spanning the back-splice junction as well as BAM files containing the mappings og all reads(alternatively of all chimeric reads)

#### Required Tools and packages 

The reconstruction module depends on bedtools (>= 2.27.0), samtools (>= 1.3.1), Python (>= 3.7; pysam> pybedtools, numpy, and pathos), and R(>= 4.0.0; amap, Hmisc, gplots). All Python an R dependencies will be installed automatically when installing circtools. Please make sure to have the correct versions of bedtools and samtools in your `$PATH`.

#### General Usage 
In order to characterize circRNAs from RNA-seq data the following steps are necessary:

    1. Mapping of RNA-seq data from quality checked FASTQ files with STAR (BWA, TopHat-Fusion in preparation)

    2. Detection circRNAs using circtools detect (CIRI, CIRCfinder or CIRCexplorer in preparation)

    3. Run circtools reconstruct

###### *Mapping of RNA-Seq data and detection of circRNAs* 
    see the documentation of `circtools detect` for instructions how to pre-process the data.


## CircTest module 

The `circtest` command is an interface for CircTest allowing for a convenient way to employ statistical testing to circRNA candidates generated with DCC without having to write an R script for each new experiment. This module allows dynamic grouping of the samples (columns) in the DCC data. 

#### Required Tools and packages 

`circtools circtest` depends on R and the following R packages:

    - aod

    - ggplot2

    - plyr

The `CircTest` R package as well as all dependencies are installed during the circtools installation procedure.

#### General Usage 
It is recommended to work with the output of the `circtools detect` module but can also run on custom count tables. Required is *one table with circRNA counts and one table with host-gene counts* that are in the same order, i.e., `circ[i,j]` and `linear[i,j]` 

A call to `circtools circtest --help` shows all available command line flags: 

```bash
usage: circtools [-h] -d DETECT_DIR -l CONDITION_LIST -c CONDITION_COLUMNS -g
                 GROUPING [-r NUM_REPLICATES] [-f MAX_FDR] [-p PERCENTAGE]
                 [-s FILTER_SAMPLE] [-C FILTER_COUNT] [-o OUTPUT_DIRECTORY]
                 [-n OUTPUT_NAME] [-m MAX_PLOTS] [-a LABEL] [-L RANGE]
                 [-O ONLY_NEGATIVE] [-H ADD_HEADER] [-M {colour,bw}]
```

##### Sample Call 
*call*
```bash
circtools circtest -d 01_detect/ -p 0.01 -s 3 -r 4 -C 2 -g 1,2,1,2,1,2,1,2 -l RNaseR-,RNaseR+ -c 4,5,6,7,8,9,10,11 -o 04_circtest/
```


## Useful links 
*Circtools* <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6795751/>
*Circtools Website* <https://docs.circ.tools/en/latest/>

